es6-arrow-functions/main.js

```diff
@@ -0,0 +1,47 @@
+const $jokeContainer = document.querySelector('#joke-container');
+const $flashImage = document.querySelector('#flash-image');
+const $jokeForm = document.querySelector('#joke-form');
+
+const jokester = {
+  tellJoke: function (setup, punchline) {
+    $jokeForm.classList.add('d-none');
+    const self = this;
+    const $introStatement = self.renderJokePhrase('Hey Flash...');
+    this.appendJokePhrase($introStatement);
+    setTimeout(function () {
+      const $jokeSetup = self.renderJokePhrase(setup);
+      self.appendJokePhrase($jokeSetup);
+      setTimeout(function () {
```

Comment:
If you convert these `setTimeout` callbacks into arrow functions, you won't have to do the `const self = this` hack.